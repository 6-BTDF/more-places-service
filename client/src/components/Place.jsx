/* eslint-disable */
import React from 'react';
import styles from './Place.module.css';
// import App from '../index.js'

const Place = ({place}) => {
  // console.log(place);
  return (
    <li className={styles.slideContainerWrapper}>
      <div className={styles.slideContainer1}>
        <div className={styles.slideContainer1_1}>
          <a className={styles.slideLink}></a>
          <div className={styles.imageSectionWrapper}>
            <div className={
              styles.imageSection
            }>
              <div className={
                styles.imageWrapper1
              }>
                <div className={
                  styles.imageWrapper1_1
                }>
                  <div className={
                  styles.imageWrapper1_1_1
                  }>
                    <img className={styles.image} src={place.pictureURL}></img>
                    <div className={
                    styles.imageWrapper1_1_1_1
                    }>
                    </div>
                  </div>
                </div>
              </div>
              <div className={styles.imageWrapper2}>
                <div className={styles.imageWrapper2_top}>
                  <div className={styles.imageWrapper2_top_right}>
                    <button type='button' className={styles.imageWrapper2_top_right_heartWrapper}>
                      <svg viewBox="0 0 32 32" className={styles.imageWrapper2_top_right_heart}>
                        <path d="m16 28c7-4.733 14-10 14-17 0-1.792-.683-3.583-2.05-4.95-1.367-1.366-3.158-2.05-4.95-2.05-1.791 0-3.583.684-4.949 2.05l-2.051 2.051-2.05-2.051c-1.367-1.366-3.158-2.05-4.95-2.05-1.791 0-3.583.684-4.949 2.05-1.367 1.367-2.051 3.158-2.051 4.95 0 7 7 12.267 14 17z">
                        </path>
                      </svg>
                    </button>
                  </div>
                </div>
                {/* <div className={styles.imageWrapper2_bottom}>
                  <div className={styles.imageWrapper2_bottom_leftside}>
                  </div>
                  <div className={styles.imageWrapper2_bottom_rightside}>
                  </div>
                  <div className={styles.imageWrapper2_bottom_bottomside}>
                  </div>
                </div> */}
              </div>
            </div>
          </div>
          <div className={styles.scoreAndReviewSectionWrapper}>
            <div className={styles.scoreAndReviewSection1}>
              <span className={styles.scoreAndReviewSection1_1}>
                <span className={styles.placeStarSectionWrapper}>
                  <svg viewBox="0 0 1000 1000" className={styles.placeStarSection}>
                    <path d="M972 380c9 28 2 50-20 67L725 619l87 280c11 39-18 75-54 75-12 0-23-4-33-12L499 790 273 962a58 58 0 0 1-78-12 50 50 0 0 1-8-51l86-278L46 447c-21-17-28-39-19-67 8-24 29-40 52-40h280l87-279c7-23 28-39 52-39 25 0 47 17 54 41l87 277h280c24 0 45 16 53 40z">
                    </path>
                  </svg>
                </span>
                <span className={styles.placeScore}>
                  {place.score.toFixed(2)}
                </span>
                <span className={styles.placeReviewCount}>
                  ({place.reviewCount})
                </span>
              </span>
            </div>
          </div>
          <div className={styles.roomAndBedSectionWrapper}>
            <ol className={styles.roomAndBedSection}>
              <li className={styles.roomSection}>
                {place.roomType}
              </li>
              <li className={styles.bedSection}>
                <span>
                  &nbsp;&middot;&nbsp;
                </span>
                {place.bedCount}
              </li>
            </ol>
          </div>
          <div className={styles.roomNameSection}>
            {place.roomName}
          </div>
          <div className={styles.costSectionWrapper}>
            <div className={styles.costSection}>
              <span className={styles.costPerNight}>
                ${place.costPerNight}
              </span>
              <span>
              &nbsp;/ night&nbsp;
              </span>
            </div>
          </div>
        </div>
      </div>
    </li>
  )
};

export default Place;
