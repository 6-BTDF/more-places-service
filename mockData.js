/* eslint-disable */
module.exports = [
  {
      "_id": "5fb5b94f7ef47dcb654487c7",
      "listingID": 4,
      "listingName": "Cassandra Brook",
      "morePlacesID": [
          {
              "_id": "5fb5b94f7ef47dcb654487c8",
              "listingID": 50,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-saviesa-home-2089698.jpg",
              "locationName": "Erdman Spring",
              "liked": false,
              "score": 5.56,
              "reviewCount": 39,
              "roomType": "Refined Soft Computer",
              "roomName": "Awesome Granite Soap",
              "bedCount": 2,
              "costPerNight": 132
          },
          {
              "_id": "5fb5b94f7ef47dcb654487ca",
              "listingID": 4,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-saviesa-home-2089698.jpg",
              "locationName": "Shanahan Bridge",
              "liked": false,
              "score": 4.24,
              "reviewCount": 66,
              "roomType": "Handmade Cotton Salad",
              "roomName": "Small Rubber Bike",
              "bedCount": 1,
              "costPerNight": 222
          },
          {
              "_id": "5fb5b94f7ef47dcb654487cc",
              "listingID": 28,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-pixabay-259588.jpg",
              "locationName": "Ullrich Inlet",
              "liked": false,
              "score": 3.22,
              "reviewCount": 159,
              "roomType": "Handmade Steel Towels",
              "roomName": "Gorgeous Soft Sausages",
              "bedCount": 1,
              "costPerNight": 179
          },
          {
              "_id": "5fb5b94f7ef47dcb654487ce",
              "listingID": 74,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/arcwind--OKp-rhSWE4-unsplash.jpg",
              "locationName": "Conner Islands",
              "liked": false,
              "score": 4.53,
              "reviewCount": 0,
              "roomType": "Licensed Fresh Chips",
              "roomName": "Rustic Metal Towels",
              "bedCount": 1,
              "costPerNight": 197
          },
          {
              "_id": "5fb5b94f7ef47dcb654487d0",
              "listingID": 44,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-tom%C3%A1%C5%A1-mal%C3%ADk-2581922.jpg",
              "locationName": "Jacobson Estates",
              "liked": false,
              "score": 3.17,
              "reviewCount": 17,
              "roomType": "Awesome Frozen Table",
              "roomName": "Handmade Rubber Fish",
              "bedCount": 2,
              "costPerNight": 105
          },
          {
              "_id": "5fb5b94f7ef47dcb654487d2",
              "listingID": 59,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-pixabay-259588.jpg",
              "locationName": "Garfield Expressway",
              "liked": false,
              "score": 1.31,
              "reviewCount": 40,
              "roomType": "Rustic Plastic Towels",
              "roomName": "Awesome Steel Car",
              "bedCount": 1,
              "costPerNight": 104
          },
          {
              "_id": "5fb5b94f7ef47dcb654487d4",
              "listingID": 97,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-tom%C3%A1%C5%A1-mal%C3%ADk-2581922.jpg",
              "locationName": "Erna Court",
              "liked": false,
              "score": 4.94,
              "reviewCount": 145,
              "roomType": "Refined Wooden Gloves",
              "roomName": "Awesome Frozen Gloves",
              "bedCount": 1,
              "costPerNight": 228
          },
          {
              "_id": "5fb5b94f7ef47dcb654487d6",
              "listingID": 20,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/arcwind--OKp-rhSWE4-unsplash.jpg",
              "locationName": "Oberbrunner Plaza",
              "liked": false,
              "score": 2.67,
              "reviewCount": 181,
              "roomType": "Rustic Granite Chicken",
              "roomName": "Handmade Concrete Chicken",
              "bedCount": 2,
              "costPerNight": 116
          },
          {
              "_id": "5fb5b94f7ef47dcb654487d8",
              "listingID": 63,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-saviesa-home-2089698.jpg",
              "locationName": "Sam Circles",
              "liked": false,
              "score": 3.63,
              "reviewCount": 130,
              "roomType": "Rustic Metal Sausages",
              "roomName": "Intelligent Soft Ball",
              "bedCount": 2,
              "costPerNight": 77
          },
          {
              "_id": "5fb5b94f7ef47dcb654487da",
              "listingID": 43,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-pixabay-259588.jpg",
              "locationName": "Pfannerstill Fall",
              "liked": false,
              "score": 1.39,
              "reviewCount": 77,
              "roomType": "Rustic Wooden Tuna",
              "roomName": "Fantastic Fresh Cheese",
              "bedCount": 2,
              "costPerNight": 142
          },
          {
              "_id": "5fb5b94f7ef47dcb654487dc",
              "listingID": 71,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/deborah-cortelazzi-gREquCUXQLI-unsplash.jpg",
              "locationName": "Blick Pass",
              "liked": false,
              "score": 4.21,
              "reviewCount": 6,
              "roomType": "Handmade Plastic Table",
              "roomName": "Small Metal Pants",
              "bedCount": 1,
              "costPerNight": 78
          },
          {
              "_id": "5fb5b94f7ef47dcb654487de",
              "listingID": 9,
              "pictureURL": "https://hrsf131-fec.s3-us-west-1.amazonaws.com/pexels-pixabay-262405.jpg",
              "locationName": "Kertzmann Plain",
              "liked": false,
              "score": 5.36,
              "reviewCount": 54,
              "roomType": "Small Concrete Shirt",
              "roomName": "Gorgeous Metal Salad",
              "bedCount": 1,
              "costPerNight": 33
          }
      ],
      "__v": 0
  }
]