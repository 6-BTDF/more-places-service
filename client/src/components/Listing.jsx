/* eslint-disable */
import React from 'react';
import Place from './Place.jsx';
import styles from './Listing.module.css';
const axios = require('axios');

class Listing extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      listing: {},
    }

    this.getListing = this.getListing.bind(this);
  }

  // get data from db on page load
  componentDidMount() {
    window.addEventListener('load', this.getListing);
  }

  getListing() {
    axios.get('/api/rooms/12')
    .then((res) => {
      // console.log(res);
      this.setState({ // this triggers a re-render
        listing: res.data[0],
      })
    })
    .catch((err) => {
      console.log(err);
    });
  }

  render() {
    const places = this.state.listing.morePlacesID;
    // console.log(places);
    // If there is nothing in this.state (because API call has not gone through),
    // do not render
    if (!places) {
      return null;
    }
    return (
      <div >
        <div className={styles.sliderContainerWrapper}>
          <div className={styles.sliderContainer}>
            <div>More places to stay</div>
            <div className={styles.sliderContainerBottom}>
              <ul className={styles.slider}>
                {places.map((place, key) => {
                  return (
                    <Place
                    key = {place.listingID}
                    place = {place}
                    />
                  )
                })}
              </ul>
            </div>

          </div>
        </div>
      </div>
    )
  }
}

export default Listing;