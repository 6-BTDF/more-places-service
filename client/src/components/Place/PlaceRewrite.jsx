/* eslint-disable */
import React from 'react';
import styled from 'styled-components';
import mockData from '../../../../mockData.js';
import SlideContainerComponent from './Slide/SlideContainerComponent.jsx';

// .slideContainerWrapper li class="_1w7e1y2"
const SlideContainerListItem = styled.li`
  border-width: 0px 10px;
  max-width: 33.3333%;
  flex: 0 0 33.3333%;
  border-style: solid;
  border-color: transparent;
  scroll-snap-align: start;
  scroll-snap-stop: always;
  font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,"Helvetica Neue",sans-serif;
  transform: translateX(-${props => props.translate * 400}px);
  transition: transform ease-out ${props => props.transition}s;
`;

class PlaceRewrite extends React.Component {

  constructor(props) {
    super(props); // gain the ability to use this.props in constructor
    const {translate, transition} = this.props;

    this.state = {
      place: this.props.place ? this.props.place : mockData[0].morePlacesID[0],
      translate: translate,
      transition: transition,
    };

    this.handleLike = this.handleLike.bind(this);
  }

  // a method to set liked to true and to be passed to Place component
  // takes in an event and an index
  handleLike(event) {
    const newState = Object.assign({}, this.state.place);
    newState.liked = !newState.liked;
    this.setState({
      place: newState
    });
  }

  render() {
    const {translate, transition} = this.props;
    return (
      <SlideContainerListItem
        translate={translate}
        transition={transition}
      >
        <SlideContainerComponent
          props={this.props}
          handleLike={this.handleLike}>
        </SlideContainerComponent>
      </SlideContainerListItem>
    )
  }
}

export default PlaceRewrite;