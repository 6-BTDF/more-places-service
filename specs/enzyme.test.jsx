/* eslint-disable */
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import Listing from '../client/src/components/Listing.jsx';
import Place from '../client/src/components/Place.jsx';
import mockData from '../mockData.js';

describe('<Listing />', () => {

  test('mounting a <Listing />', () => {
    const testComponent = TestRenderer.create(
      <Listing />
    );

    let tree = testComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  const wrapper = mount(<Listing />);

  test('should not render <Place /> with no data', () => {
    expect(wrapper.find(Place)).toHaveLength(0);
  })

  test('should render <Place /> with data', () => {
    wrapper.setState({listing: mockData[0]});
    expect(wrapper.find(Place)).toHaveLength(12);
  })

  test('should contain ul', () => {
    wrapper.setState({listing: mockData[0]});
    expect(wrapper.exists('ul')).toBeTruthy();
  })
})

describe('<Place />', () => {

  const wrapper = mount(<Place />);
  wrapper.setState({place: mockData[0].morePlacesID[0]});

  test('should contain li', () => {
    expect(wrapper.exists('li')).toBeTruthy();
  })

  test('should contain ol', () => {
    expect(wrapper.exists('ol')).toBeTruthy();
  })
});
