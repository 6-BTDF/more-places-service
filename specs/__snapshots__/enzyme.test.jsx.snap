// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listing /> mounting a <Listing /> 1`] = `null`;
